### PR TITLE
Cookbook Recipe - Onboarding

### PR DESCRIPTION
**Pull request by Loquacity**
# Description

Draft of onboarding recipe.

# Target branches

Which documentation version does this PR apply to?

- [x] Master (Default)
- [ ] Manager-4.1
- [ ] Manager-4.0

# Links

Partial https://github.com/SUSE/spacewalk/issues/14304
Epic https://github.com/SUSE/spacewalk/issues/12021

### CODE REVIEW

File: modules/cookbook/pages/recipe-onboarding.adoc [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+== Problem
+
+This recipe shows you how to onboard a client to your {productname} Server.
+
+The client can be running any Linux operating system.
+For a list of supported client systems, see xref:installation:client-requirements.adoc[].
+
+This is one of the first tasks you need to do when you set up {productname} for the first time, and you will probably have to do it many more times as you use the product.
+
+This recipe shows you how to onboard a single client, for a recipe about onboarding multiple clients at the same time, see xref:FIXME[].
```
Comment (keichwa) [inline modules/cookbook/pages/recipe-onboarding.adoc:19]:
```suggestion
```
This xref is distracting.  If needed we can add it at the end of this chapter, imo.

File: modules/cookbook/pages/recipe-onboarding.adoc [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+This recipe uses a {sles} 15 SP2 operating system.
+You can use other Linux operating systems, but some of the steps might be different.
+For more information on onboarding other clients, see xref:client-configuration:registration-methods.adoc[].
+
+
+
+== Steps
+
+Configure a fully-qualified domain name (FQDN) on your client::
```
Comment (keichwa) [inline modules/cookbook/pages/recipe-onboarding.adoc:46]: The trailing two colons seem to be intentional.

File: modules/cookbook/pages/recipe-onboarding.adoc [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+* On the client, at the command prompt, show the current hostname:
++
+----
+hostname -f
+----
++
+This command will probably return an error, or show something like ``localhost``.
+* Set a new hostname.
+  Your new hostname should include two periods.
```
Comment (keichwa) [inline modules/cookbook/pages/recipe-onboarding.adoc:56]:
```suggestion
  Your new hostname should have a subdomain name and thus include at least two periods.
```

File: modules/cookbook/pages/recipe-onboarding.adoc [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+hostname -f
+----
++
+This command should return your new FQDN.
+
+Prepare software channels on the {productname} Server::
+
+* In the {productname} {webui}, navigate to menu:Admin[Setup Wizard].
+* In the [guimenu]``Organization Credentials`` tab, ensure you have entered your {scc} credentials, and are correctly authenticated.
+* In the [guimenu]``Products`` tab, ensure that the product catalog is fully updated:
```
Comment (keichwa) [inline modules/cookbook/pages/recipe-onboarding.adoc:83]: I wouldn't introduce the new term "product catalog" (or is the that used in the UI?).

File: modules/cookbook/pages/recipe-onboarding.adoc [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
++
+image::scc_sync.png[scaledwidth=80%]
+* Use the product search bar to find the channels you need for your client operating system.
```
Comment (keichwa) [inline modules/cookbook/pages/recipe-onboarding.adoc:86]: Is this actually a separate step?

File: modules/cookbook/pages/recipe-onboarding.adoc [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+Check the channels you want to install, and click btn:[Add products]:
```
Comment (keichwa) [inline modules/cookbook/pages/recipe-onboarding.adoc:87]: here, indentation probably is missing.

File: modules/cookbook/pages/recipe-onboarding.adoc [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
++
+image::add_products.png[scaledwidth=80%]
+* Wait for the products to fully synchronize.
```
Comment (keichwa) [inline modules/cookbook/pages/recipe-onboarding.adoc:90]:
```suggestion
* Wait for the product channels to fully synchronize.
```

File: modules/cookbook/pages/recipe-onboarding.adoc [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+  Depending on the products you have chosen, this could take a long time.
+
+
+Create an activation key::
+
+* In the {productname} {webui}, navigate to menu:Systems[Activation Keys], and click btn:[Create Key].
+* Give your activation key a name, and select the base channel that matches the client you want to onboard.
+This should be the product you enabled in the last step:
```
Comment (keichwa) [inline modules/cookbook/pages/recipe-onboarding.adoc:98]: "procedure" instead of "step"?

File: modules/cookbook/pages/recipe-onboarding.adoc [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
++
+image::activationkey_base.png[scaledwidth=80%]
+* Check the child channels to include, and any add-on system types you want clients registered with this key to have.
+  If you want this activation key to be used for all clients by default, check [guimenu]``Universal Default``.
```
Comment (keichwa) [inline modules/cookbook/pages/recipe-onboarding.adoc:102]: I think we'd better warn about this feature.

**Review by keichwa — CHANGES_REQUESTED** [review]

File: modules/cookbook/pages/recipe-onboarding.adoc (hunk 2 above)
Comment (Loquacity) [reply modules/cookbook/pages/recipe-onboarding.adoc:46]: yep, to turn it into a deflist.

File: modules/cookbook/pages/recipe-onboarding.adoc (hunk 4 above)
Comment (Loquacity) [reply modules/cookbook/pages/recipe-onboarding.adoc:83]: I think that's from the web UI, but I can double check.

**Review by keichwa — APPROVED** [review]

File: modules/cookbook/pages/recipe-onboarding.adoc (hunk 4 above)
Comment (Loquacity) [reply modules/cookbook/pages/recipe-onboarding.adoc:82]: image

File: modules/cookbook/pages/recipe-onboarding.adoc (hunk 5 above)
Comment (Loquacity) [reply modules/cookbook/pages/recipe-onboarding.adoc:85]: This step includes entering a search term, checking the boxes, and clicking 'add products'. So I think that's a logical grouping.

File: modules/cookbook/pages/recipe-onboarding.adoc (hunk 9 above)
Comment (Loquacity) [reply modules/cookbook/pages/recipe-onboarding.adoc:101]: what do you mean?

File: modules/cookbook/pages/recipe-onboarding.adoc [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+
+Prepare software channels on the {productname} Server::
+
+* In the {productname} {webui}, navigate to menu:Admin[Setup Wizard].
+* In the [guimenu]``Organization Credentials`` tab, ensure you have entered your {scc} credentials, and are correctly authenticated.
+* In the [guimenu]``Products`` tab, ensure that the product catalog is fully updated:
++
+image::scc_sync.png[scaledwidth=80%]
+* Use the product search bar to find the channels you need for your client operating system.
+Check the channels you want to install, and click btn:[Add products]:
```
Comment (Loquacity) [inline modules/cookbook/pages/recipe-onboarding.adoc:86]:
```suggestion
  Check the channels you want to install, and click btn:[Add products]:
```

File: modules/cookbook/pages/recipe-onboarding.adoc (hunk 8 above)
Comment (Loquacity) [reply modules/cookbook/pages/recipe-onboarding.adoc:97]: What about this:

```suggestion
This should be the product you just enabled:
```

Or, maybe take it out entirely?

File: modules/cookbook/pages/recipe-onboarding.adoc (hunk 9 above)
Comment (keichwa) [reply modules/cookbook/pages/recipe-onboarding.adoc:101]: AFAIK, it is not recommended to make use of this feature.  Thus, here in the context of the cookbook, we'd better ask the user to keep it unchecked.

**Review by keichwa — APPROVED** [review]

File: modules/cookbook/pages/recipe-onboarding.adoc (hunk 9 above)
Comment (Loquacity) [reply modules/cookbook/pages/recipe-onboarding.adoc:101]: OK, I'll just take the sentence out, in that case.

File: modules/cookbook/pages/recipe-onboarding.adoc [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+
+Create an activation key::
+
+* In the {productname} {webui}, navigate to menu:Systems[Activation Keys], and click btn:[Create Key].
+* Give your activation key a name, and select the base channel that matches the client you want to onboard.
+This should be the product you just enabled:
++
+image::activationkey_base.png[scaledwidth=80%]
+* Check the child channels to include, and any add-on system types you want clients registered with this key to have.
+  If you want this activation key to be used for all clients by default, check [guimenu]``Universal Default``.
```
Comment (Loquacity) [inline modules/cookbook/pages/recipe-onboarding.adoc:101]:
```suggestion
```

**Pull request merged**